task_processing/plugins/mesos/retrying_executor.py

```diff
@@ -0,0 +1,60 @@
+import time
+from threading import Thread
+
+from pyrsistent import m
+from six.moves.queue import Queue
+
+from task_processing.interfaces.task_executor import TaskExecutor
+
+
+class RetryingExecutor(TaskExecutor):
```

Comment:
I'd kind of expect this behavior to be at the ``Runner`` layer, rather than ``Executor``? (I'd expect this to have looked very similar to the ``AsyncRunner``, with the ``predicate`` matching on non-success && terminal events, and the ``cb`` ``executor.run``)?